src/Gui/Tree.cpp

```diff
@@ -855,19 +860,35 @@ void TreeWidget::contextMenuEvent (QContextMenuEvent * e)
         DocumentObjectItem* objitem = static_cast<DocumentObjectItem*>
             (this->contextItem);
 
+        // check that the selection is not across several documents
+        bool acrossDocuments = false;
+        auto SelectedObjectsList = Selection().getCompleteSelection();
+        QList<QTreeWidgetItem*> items = this->selectedItems();
+        // get the object's document as reference
         App::Document* doc = objitem->object()->getObject()->getDocument();
+        for (auto it = SelectedObjectsList.begin(); it != SelectedObjectsList.end(); ++it) {
+            if ((*it).pDoc != doc)
+                acrossDocuments = true;

```

Comment:
You could break the loop here, I think even [Core Guideline ES.77](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es77-minimize-the-use-of-break-and-continue-in-loops) would consider this case trivial.

Reply:
indeed, now done